lib/logstash/inputs/unix.rb

```diff
@@ -61,6 +66,9 @@ def register
                       :path => @path)
         raise
       end
+      if not @chmod.nil?
```

Comment:
I would recommend using something like ```File.chmod(@chmod, @path) if @chmod``` as it keeps the syntax more clear as the one used.